ml-algorithms/src/main/java/org/opensearch/ml/engine/ingest/OpenAIDataIngestion.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.engine.ingest;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.security.AccessController;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.opensearch.OpenSearchStatusException;
+import org.opensearch.client.Client;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.ml.common.transport.batch.MLBatchIngestionInput;
+import org.opensearch.ml.engine.annotation.Ingester;
+
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+@Ingester("openai")
+public class OpenAIDataIngestion extends AbstractIngestion {
+    private static final String API_KEY = "openAI_key";
+    private static final String API_URL = "https://api.openai.com/v1/files/";
```

Comment:
Hard coded URL makes it hard to extend. If OpenAI release v2 API url, we have to make code change

Reply:
I'm planning to make this ingester general enough for all file servers, but the interphase and UX cases remain the same. So let's leave this enhancement in the refactor as a separate PR. It may takes some time for more tests.